packages/extension-utils/src/extensionAdaptor.ts

```diff
@@ -59,6 +59,12 @@ export class ExtensionAdaptor
     )
   }
 
+  async copyToClipboard(location: { pathname: string; search: string }) {
```

Comment:
Commenting to make visible to @josephaxisa @bryans99 : Per my discussion with Bryn today, I need a way to access the browser window.location object in this adaptor in order to append its pathname/search params to the hostname/url contained by the extensionSdk. I do this by passing in the `location` returned from `useLocation` when called in CopyLinkWrapper.

I experimented with a couple of different approaches for passing in the parameter, but settled on this typed object. This made it easier than dealing with the location object overall and its type issues when passing it in through the spec file as opposed to after getting it from the `useLocation` hook in CopyLinkWrapper. Thoughts on this approach?

Reply:
Change to this. Route is available in lookerHostData. No need to pass it in
```

  async copyToClipboard() {
    const routeContext = this.extensionSdk.lookerHostData
    const { lookerHostData } = this.extensionSdk
    if (lookerHostData) {
      const { extensionId, route } = lookerHostData
      const url = `${routeContext?.hostUrl}/extensions/${extensionId}${route}`
      await this.extensionSdk.clipboardWrite(url)
    }
  }
```

Update `packages/extension-utils/src/adaptorUtils.ts` as follows

```
export interface IEnvironmentAdaptor extends IAuthAdaptor {
  /** Method for determining whether running in a browser or extension environment */
  isExtension(): boolean
  /** Method for retrieving a keyed value from local storage */
  localStorageGetItem(key: string): Promise<string | null>
  /** Method for setting a keyed value in local storage */
  localStorageSetItem(key: string, value: string): void
  /** Method for removing a keyed value from local storage */
  localStorageRemoveItem(key: string): void
  /** Theme settings for extension */
  themeOverrides(): ThemeOverrides
  /** Open a new browser window with the given url and target  */
  openBrowserWindow: (url: string, target?: string) => void
  /** error logger */
  logError: (error: Error, componentStack: string) => void
  /** copy to clipboard */
  copyToClipboard: () => Promise<void>
}
```

And CopyLinkWrapper.tsx as follows
```
  const handleCopyLink = async () => {
    await getEnvAdaptor().copyToClipboard()
    // await navigator.clipboard.writeText(location.href)
    setTooltipContent('Copied to clipboard')
  }
```